images/base/files/usr/local/bin/entrypoint

```diff
@@ -203,20 +203,47 @@ fix_cgroup() {
     return
   fi
   echo 'INFO: detected cgroup v1'
-  echo 'INFO: fix cgroup mounts for all subsystems'
   # See: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
   # Capture initial state before modifying
   #
   # Basically we're looking for the cgroup-path for the cpu controller for the
   # current process. this tells us what cgroup-path the container is in.
   # Then we collect the subsystems that are active on this path.
-  # We assume the cpu controller is in use on all node containers.
+  # We assume the cpu controller is in use on all node containers,
+  # and other controllers use the same sub-path.
   #
   # See: https://man7.org/linux/man-pages/man7/cgroups.7.html
   local current_cgroup
   current_cgroup=$(grep -E '^[^:]*:([^:]*,)?cpu(,[^,:]*)?:.*' /proc/self/cgroup | cut -d: -f3)
+  if [ "$current_cgroup" = "/" ]; then
+    echo "INFO: cgroupns detected, no need to fix cgroups"
```

Comment:
For simplicity I'd suggest always enabling `--cgroupns=private`.
This is not present in Docker 19.03 CLI, but 19.03 has already reached EOL.

Reply:
I concur (see https://github.com/kubernetes-sigs/kind/pull/2709#issuecomment-1092350339 which I wrote yesterday but forgot to click "Comment").

Reply:
So, we can document that KIND now requires `--cgroupns=private`, and replace the whole `fix_cgroup` with a check that we got a private cgroupns.

Reply:
I'm not sure though it if will work for other environments. It does work for podman (checked with a version bundled with Ubuntu 21.10). The script also mentions "Cloud Shell" (not sure what's that but seems to be k8s based) and GitHub Actions.

In fact, I think I know what to do. Please see the last commit (just updated).